skipPom is ignored by "push" goal

<!-- Please fill out the following information to help us in analyzing the issue, but feel free to skip it if you don't have the information at hand or if it does not apply. Please remove everything which does not apply to the issue. -->

### Description
Have a multi-module project with docker-maven-plugin configuration in the images-parent module, which has packaging "pom".
Each subdirectory module has a simple minimal pom.xml with packaging "jar" and Dockerfile to build an image.
The d-m-p configuration has skipPom=true configured
The "build" goal is bound to "package" amd "push" is bound to "install".

During the "build" goal, the pom with packaging=pom is correctly skipped as expected.
unfortunately the "push" goal ignores skipPom and attempts to run in the parent pom.

```
[INFO] --- docker-maven-plugin:0.36.0:build (docker-build) @  images-parent  ---
[INFO] DOCKER> [ images-parent :1-0-0-snapshot] : Skipped building
[INFO]

[INFO] --- docker-maven-plugin:0.36.0:push (repository-push) @ images-parent ---
[ERROR] DOCKER> Unable to add tag [our.registry.com/docker/images/images-parent:1-0-0-snapshot] to image [images-parent:1-0-0-snapshot] : {"message":"No such image: images-parent:1-0-0-snapshot"} (Not Found: 404) [{"message":"No such image: images-parent:1-0-0-snapshot"} (Not Found: 404)]
```
### Info

* docker-maven-plugin version :0.36.0
* Maven version (`mvn -v`) :
```
Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)
Maven home: ~/Apps/apache-maven-3.8.1
Java version: 11.0.10, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
Default locale: en_GB, platform encoding: UTF-8
OS name: "mac os x", version: "10.16", arch: "x86_64", family: "mac"
```
* Docker version :
```
Client:
 Cloud integration: 1.0.17
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.16.4
 Git commit:        f0df350
 Built:             Wed Jun  2 11:56:22 2021
 OS/Arch:           darwin/amd64
 Context:           desktop-linux
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       b0f5bc3
  Built:            Wed Jun  2 11:54:58 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.6
  GitCommit:        d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc:
  Version:          1.0.0-rc95
  GitCommit:        b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
* If it's a bug, how to reproduce :
In images-parent:
```
	<modules>
		<module>child</module>
	</modules>

	<build>
		<plugins>
			<plugin>
				<groupId>io.fabric8</groupId>
				<artifactId>docker-maven-plugin</artifactId>
				<extensions>true</extensions>
				<executions>
					<execution>
						<id>docker-build</id>
						<phase>package</phase>
						<goals>
							<goal>build</goal>
						</goals>
						<configuration>
							<images>
								<image>
									<name>%a:${docker.release.tag}</name>
									<build>
										<contextDir>${docker.contextDir}</contextDir>
										<assembly>
											<name>dependencies</name>
											<descriptorRef>dependencies</descriptorRef>
										</assembly>
										<cleanup>try</cleanup>
										<filter>@</filter>
									</build>
								</image>
							</images>
						</configuration>
					</execution>
					<execution>
						<id>repository-push</id>
						<phase>install</phase>
						<goals>
							<goal>push</goal>
						</goals>
						<configuration>
							<images>
								<image>
									<name>%a:${docker.release.tag}</name>
									<build/>
								</image>
							</images>
						</configuration>
					</execution>
				</executions>
				<configuration>
					<dockerHost>${docker.host}</dockerHost>
					<certPath>${docker.certPath}</certPath>
					<sourceDirectory>.</sourceDirectory>
					<imagePullPolicy>IfNotPresent</imagePullPolicy>
					<pushRegistry>${docker.target.repository}</pushRegistry>
					<authConfig>
						<username>${docker.auth.user}</username>
						<password>${docker.auth.pass}</password>
					</authConfig>
					<retries>1</retries>
					<skipPush>${docker.skip.push}</skipPush>
					<skipPom>true</skipPom>
					<useColor>true</useColor>
					<verbose>true</verbose>
				</configuration>
			</plugin>
		</build>
```
 
The pom for "child"  has minimal configuration, the idea is it's inheriting all build config from its parent.

The workaround for the moment is to put everything in the parent into pluginManagement and add a build section to every child:
```
	<build>
		<plugins>
			<plugin>
				<groupId>io.fabric8</groupId>
				<artifactId>docker-maven-plugin</artifactId>
			</plugin>
		</plugins>
	</build>
```
```

* If it's a feature request, what is your use case :
* Sample project : *[GitHub Clone URL]*

--- a/doc/changelog.md
+++ b/doc/changelog.md
@@ -1,6 +1,7 @@
 # ChangeLog
 
 * **0.39-SNAPSHOT** :
+  - `skipPom` is ignored by "push" goal ([1482](https://github.com/fabric8io/docker-maven-plugin/issues/1482))
   - Cleanup dangling images as a result of image tagging, auto-pulling a base image, or auto-pulling a cacheFrom image ([#1513](https://github.com/fabric8io/docker-maven-plugin/pull/1513)) @rkhmelichek
   - Enable Create Image (pull) HTTP API option ([#1516](https://github.com/fabric8io/docker-maven-plugin/issues/1516)) @rohanKanojia
 

--- a/src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java
+++ b/src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java
@@ -222,6 +222,12 @@ public abstract class AbstractDockerMojo extends AbstractMojo implements Context
     @Parameter
     private DockerMachineConfiguration machine;
 
+    @Parameter(defaultValue = "${project.packaging}", required = true)
+    protected String packaging;
+
+    @Parameter(property = "docker.skip.pom", defaultValue = "false")
+    protected boolean skipPom;
+
     // Images resolved with external image resolvers and hooks for subclass to
     // mangle the image configurations.
     private List<ImageConfiguration> resolvedImages;
@@ -576,6 +582,10 @@ public abstract class AbstractDockerMojo extends AbstractMojo implements Context
         ImagePullManager pullManager = getImagePullManager(determinePullPolicy(runConfiguration), autoPull);
         RegistryConfig registryConfig = getRegistryConfig(pullRegistry);
         registryService.pullImageWithPolicy(imageName, pullManager, registryConfig, imageConfig.getBuildConfiguration());
+    }
+
+    protected boolean shouldSkipPom() {
+        return skipPom && packaging.equalsIgnoreCase("pom");
     }
 
     private boolean containerMatchesPattern(Container container, Matcher imageNameMatcher, Matcher containerNameMatcher,

--- a/src/main/java/io/fabric8/maven/docker/BuildMojo.java
+++ b/src/main/java/io/fabric8/maven/docker/BuildMojo.java
@@ -39,14 +39,8 @@ public class BuildMojo extends AbstractBuildSupportMojo {
     @Parameter(property = "docker.skip.build", defaultValue = "false")
     protected boolean skipBuild;
 
-    @Parameter(property = "docker.skip.pom", defaultValue = "false")
-    protected boolean skipPom;
-
     @Parameter(property = "docker.name", defaultValue = "")
     protected String name;
-
-    @Parameter(defaultValue = "${project.packaging}", required = true)
-    protected String packaging;
 
     /**
      * Skip Sending created tarball to docker daemon
@@ -151,7 +145,7 @@ public class BuildMojo extends AbstractBuildSupportMojo {
         BuildImageConfiguration buildConfig = aImageConfig.getBuildConfiguration();
 
         if (buildConfig != null) {
-            if(buildConfig.skip() || (skipPom && packaging.equalsIgnoreCase("pom"))) {
+            if (buildConfig.skip() || shouldSkipPom()) {
                 log.info("%s : Skipped building", aImageConfig.getDescription());
             } else {
                 buildAndTag(hub, aImageConfig);

--- a/src/main/java/io/fabric8/maven/docker/PushMojo.java
+++ b/src/main/java/io/fabric8/maven/docker/PushMojo.java
@@ -41,7 +41,7 @@ public class PushMojo extends AbstractDockerMojo {
      */
     @Override
     public void executeInternal(ServiceHub hub) throws DockerAccessException, MojoExecutionException {
-        if (skipPush) {
+        if (skipPush || shouldSkipPom()) {
             return;
         }
 

--- a/src/test/java/io/fabric8/maven/docker/PushMojoTest.java
+++ b/src/test/java/io/fabric8/maven/docker/PushMojoTest.java
@@ -1,0 +1,112 @@
+package io.fabric8.maven.docker;
+
+import io.fabric8.maven.docker.access.DockerAccessException;
+import io.fabric8.maven.docker.config.ImageConfiguration;
+import io.fabric8.maven.docker.service.RegistryService;
+import mockit.Deencapsulation;
+import mockit.Mocked;
+import mockit.Tested;
+import mockit.Verifications;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Collection;
+
+public class PushMojoTest extends BaseMojoTest {
+  @Tested
+  private PushMojo pushMojo;
+
+  @Mocked
+  private RegistryService registryService;
+
+  @Test
+  public void executeInternal_whenSkipPomEnabledInPomPackaging_thenImagePushSkipped() throws MojoExecutionException, IOException {
+    Deencapsulation.setField(serviceHub, "registryService", registryService);
+    givenMavenProject(pushMojo);
+    givenPackaging("pom");
+    givenSkipPom(true);
+
+    whenMojoExecutes();
+
+    thenImageNotPushed();
+  }
+
+  @Test
+  public void executeInternal_whenPomPackaging_thenImageIsPushed() throws MojoExecutionException, IOException {
+    Deencapsulation.setField(serviceHub, "registryService", registryService);
+    givenMavenProject(pushMojo);
+    givenPackaging("pom");
+    givenSkipPom(false);
+
+    whenMojoExecutes();
+
+    thenImagePushed();
+  }
+
+  @Test
+  public void executeInternal_whenJarPackaging_thenImageIsPushed() throws MojoExecutionException, IOException {
+    Deencapsulation.setField(serviceHub, "registryService", registryService);
+    givenMavenProject(pushMojo);
+    givenPackaging("jar");
+
+    whenMojoExecutes();
+
+    thenImagePushed();
+  }
+
+  @Test
+  public void executeInternal_whenSkipEnabled_thenImageIsPushed() throws MojoExecutionException, IOException {
+    Deencapsulation.setField(serviceHub, "registryService", registryService);
+    givenMavenProject(pushMojo);
+    givenPackaging("jar");
+    givenSkipPush(true);
+
+    whenMojoExecutes();
+
+    thenImageNotPushed();
+  }
+
+  @Test
+  public void executeInternal_whenSkipDisabled_thenImageIsPushed() throws MojoExecutionException, IOException {
+    Deencapsulation.setField(serviceHub, "registryService", registryService);
+    givenMavenProject(pushMojo);
+    givenPackaging("jar");
+    givenSkipPush(false);
+
+    whenMojoExecutes();
+
+    thenImagePushed();
+  }
+
+  private void thenImagePushed() throws MojoExecutionException, DockerAccessException {
+    new Verifications() {{
+      registryService.pushImages((Collection<ImageConfiguration>) any, anyInt, (RegistryService.RegistryConfig)any, anyBoolean);
+      times = 1;
+    }};
+  }
+
+  private void thenImageNotPushed() throws DockerAccessException, MojoExecutionException {
+    new Verifications() {{
+      registryService.pushImages((Collection<ImageConfiguration>) any, anyInt, (RegistryService.RegistryConfig)any, anyBoolean);
+      times = 0;
+    }};
+  }
+
+  private void whenMojoExecutes() throws IOException, MojoExecutionException {
+    pushMojo.executeInternal(serviceHub);
+  }
+
+  private void givenPackaging(String packaging) {
+    Deencapsulation.setField(pushMojo, "packaging", packaging);
+  }
+
+  private void givenSkipPom(boolean skipPom) {
+    Deencapsulation.setField(pushMojo, "skipPom", skipPom);
+  }
+
+  private void givenSkipPush(boolean skip) {
+    Deencapsulation.setField(pushMojo, "skipPush", skip);
+  }
+
+}

